File: src/java/foundation/pEp/jniadapter/PassphraseEntry.java
package foundation.pEp.jniadapter;

public class PassphraseEntry {
    public String email;
    public String passphrase;

    public PassphraseEntry(_PassphraseEntry entry) {
        email = Utils.toUTF16(entry.email);
        passphrase = Utils.toUTF16(entry.passphrase);
    }
}

File: src/cxx/passphrase_entry.cc
#include "passphrase_entry.hh"

namespace pEp {
    namespace JNIAdapter {
        passphrase_entry::passphrase_entry(const std::string email, const std::string passphrase) :
                email { email, 0, passphrase_entry::max_len },
                passphrase{ passphrase, 0, passphrase_entry::max_len }
        {
        }
    }
}
